src/mixer/basetrackplayer.cpp

```diff
@@ -168,21 +171,31 @@ void BaseTrackPlayerImpl::loadTrack(TrackPointer pTrack) {
 
     // The loop in and out points must be set here and not in slotTrackLoaded
     // so LoopingControl::trackLoaded can access them.
-    const QList<CuePointer> trackCues(m_pLoadedTrack->getCuePoints());
-    QListIterator<CuePointer> it(trackCues);
-    while (it.hasNext()) {
-        CuePointer pCue(it.next());
-        if (pCue->getType() == Cue::LOOP) {
-            double loopStart = pCue->getPosition();
-            double loopEnd = loopStart + pCue->getLength();
-            if (loopStart != kNoTrigger && loopEnd != kNoTrigger && loopStart <= loopEnd) {
-                m_pLoopInPoint->set(loopStart);
-                m_pLoopOutPoint->set(loopEnd);
-                break;
+    if (!m_cloneFromChannel) {
+        const QList<CuePointer> trackCues(m_pLoadedTrack->getCuePoints());
+        QListIterator<CuePointer> it(trackCues);
+        while (it.hasNext()) {
+            CuePointer pCue(it.next());
+            if (pCue->getType() == Cue::LOOP) {
+                double loopStart = pCue->getPosition();
+                double loopEnd = loopStart + pCue->getLength();
+                if (loopStart != kNoTrigger && loopEnd != kNoTrigger && loopStart <= loopEnd) {
+                    m_pLoopInPoint->set(loopStart);
+                    m_pLoopOutPoint->set(loopEnd);
+                    break;
+                }
             }
         }
+    } else {
+        // copy loop in and out points from other deck because any new loops
+        // won't be saved yet
+        m_pLoopInPoint->set(ControlObject::get(
```

Comment:
For now this is okay, but in the long term I'd like to move away from using ControlObjects to access values from random other places in the code. This makes it difficult to know what code could be affected whenever someone next changes code that impacts the CO.

Reply:
What is the alternative here?